File: client/packages/types/validation/hospitalization-add.ts
import { z } from "zod";

export const hospitalizationAddSchema = z.object({
	patient_id: z.string().nonempty(),
	admission_date: z.string().nonempty(),
	related_to: z.string().nonempty(),
	status: z.string().nonempty(),
	length_of_stay: z.number().nullable(),
	procedure: z.string().nonempty(),
	comments: z.string(),
});

File: server/patient/src/dto/history/hospitalization/hospitalization_add.rs
use chrono::{DateTime, Utc};
use common::dto::last_updated_input::LastUpdatedInput;
use serde::{Deserialize, Serialize};
use ts_rs::TS;
use validator::Validate;

#[derive(Deserialize, Serialize, Debug, Clone, Validate, Default, PartialEq, TS, Eq)]
#[ts(export, export_to = "../../bindings/HospitalizationAdd.ts")]
pub struct HospitalizationAdd {
    pub patient_id: String,
    pub admission_date: DateTime<Utc>,
    pub related_to: Option<String>,
    pub status: Option<String>,
    pub length_of_stay: Option<u32>,
    pub procedure: Option<String>,
    pub comments: Option<String>,
    pub last_updated_input: LastUpdatedInput,
}
